llvm/cmake/modules/LLVMProcessSources.cmake

```diff
@@ -58,14 +58,16 @@ function(llvm_process_sources OUT_VAR)
   set(sources ${ARG_UNPARSED_ARGUMENTS})
   llvm_check_source_file_list(${sources})
 
-  foreach(fn ${sources})
-    get_filename_component(suf ${fn} EXT)
-    if("${suf}" STREQUAL ".cpp" OR "${suf}" STREQUAL ".c")
-      get_filename_component(short_name ${fn} NAME)
-      set_source_files_properties(${fn} PROPERTIES COMPILE_DEFINITIONS "__SHORT_FILE__=\"${short_name}\"")
-    endif()
-  endforeach()
-
+  # Don't generate __SHORT_FILE__ on MSVC builds as it can force repeated cache regeneration.
```

Comment:
Can you provide more info on the "repeated cache generation" aspect? You mentioned before that the build would come to a halt which seems a bit different than a cache problem?
At least link to the discussion on the original PR maybe?

It's not clear to me that someone reading this right now would even connect the dots to MSBuild-specific issue.

Reply:
I mean you refer to a "cache", which cache? In general I think of "ccache" when I read this. If you mean that this triggers repeated cmake invocation, then why not write it as is?
You likely should be able to see which process is churning why things seem slow?  If cmake is re-running that should be visible, and there should be no compilation job running concurrently?

Reply:
The update message may be good enough for now, I hope this isn't setting up a Chesterton's Fence though!